src/HL7/Message.php

```diff
@@ -119,16 +119,33 @@ public function __construct(string $msgStr = null, array $hl7Globals = null, boo
                     }
 
                     $preg_flags = $keepEmptySubFields ? 0 : PREG_SPLIT_NO_EMPTY;
-                    $comps = preg_split("/\\" . $this->componentSeparator .'/', $fields[$j], -1, $preg_flags);
-
-                    foreach ($comps as $k => $kValue) {
-                        $subComps = preg_split("/\\" . $this->subcomponentSeparator . '/', $comps[$k]);
-
-                        // Make it a ref or just the value
-                        (\count($subComps) === 1) ? ($comps[$k] = $subComps[0]) : ($comps[$k] = $subComps);
+                    if(stripos($fields[$j],$this->repetitionSeparator)==false) {
```

Comment:
I think `strpos` should do here since we aren't dealing with any alphabets
Should we perhaps use `===` instead of `==` to make it stricter? 
Considering above and some PSR-2 spacing, I think this could be updated to:
```php
if (strpos($fields[$j], $this->repetitionSeparator) === false) {
```